### DISCUSSION TITLE
Toggle Button Validation

### DISCUSSION THREAD

**Discussion by mclamanna** [twbs/bootstrap#30608] [General] [answered]
Is it possible to have validation on toggle buttons?
<img width="581" alt="Screen Shot 2020-04-16 at 2 10 39 PM" src="https://user-images.githubusercontent.com/29867114/79491188-11734d80-7fec-11ea-8031-3b8231803936.png">

**Comment by mdo** [comment, marked as answer]
There's no validation styles for this currently, no. I'm not sure what we'd be able to do for that with CSS alone that'd be acceptable for all situations. I'm open to ideas!

One idea for me is, if those are truly radio powered buttons, you could theoretically use them to toggle form validation messages. I just don't know about how, if at all, to restyle the buttons.

**Reply by mclamanna** [reply]
We found a solution, basically taking this:
` <input type="radio" name="suggestion" id="Validationoption1" required="">Suggestion`
and making these small changes:
` <input type="radio" name="radio" value="suggestion" id="Validationoption1" required="">Suggestion`

[Here is my pen showing a working example.](https://codepen.io/mclamanna/pen/QWjNbML)

<img width="540" alt="Screen Shot 2020-04-17 at 9 51 24 PM" src="https://user-images.githubusercontent.com/29867114/79625246-aca41a00-80f5-11ea-805f-114d55271449.png">

<img width="523" alt="Screen Shot 2020-04-17 at 9 51 35 PM" src="https://user-images.githubusercontent.com/29867114/79625250-af067400-80f5-11ea-984f-a3259e4900c7.png">

Thank you for making Bootstrap!

**Reply by ffoodd** [reply]
FWIW, using this solution I think we should be able to re-style buttons by placing hidden radio **before** button group and using something like `[type="radio"]:invalid ~ .btn-group`.

**Reply by mclamanna** [reply]
Forgot to add this:
`<input type="radio" class="custom-control-input" id="customControlValidation3" name="radio" style="display:none;" required="">
								<div class="invalid-feedback">
									Please select your topic.
								</div>`
